src/common/temporary_file.cpp

```diff
@@ -0,0 +1,62 @@
+#include "common/temporary_file.hpp"
+#include <cerrno>
+
+namespace mesos {
+namespace criteo {
+
+/*
+ * Represent a temporary file that can be either written or read from.
+ */
+TemporaryFile::TemporaryFile() {
+  char filepath[] = "/tmp/criteo-mesos-XXXXXX";
+  int fd = mkstemp(filepath);
```

Comment:
Why not use stout's `mktemp()`? It will simply the code a little bit and better integrate with Mesos' codebase
see https://github.com/criteo-forks/mesos/blob/criteo/1.8.1/3rdparty/stout/include/stout/os/posix/mktemp.hpp#L36-L58